CONTRIBUTING.md

```diff
@@ -0,0 +1,119 @@
+   
+   - Utilize Scikit-learn or similar
+   
+   Scikit-learn models tend to be fast, but usually have lower accuracy than deep learning methods. However, for well defined problems with well defined features, they can provide very good results.
+   Since deep learning models tend to be complex and slow, we encourage you to first test a traditional approach (like Conditional Random Fields) before going directly into state-of-the-art *Sesame-Street* character based models... 
+   When building such model, you would need to extend the [EntityRecognizer](presidio-analyzer/analyzer/entity_recognizer.py) class.
+
+   - Use custom logic
+
+   In some cases, rule-based logic provides the best way of detecting entities. The Presidio EntityRecognizer API allows you to use spaCy extracted features like lemmas, part of speech, dependencies and more to create your logic. When building such model, you would need to extend the [EntityRecognizer](presidio-analyzer/analyzer/entity_recognizer.py) class.
+
+   - Deep learning based methods
+
+   Deep learning methods offer excellent detection rates for NER. They are however more complex to train, deploy and tend to be slower than traditional approaches. When contributing a DL based method, the best option would be to create a sidecar container which is isolated from the presidio-analyzer container. On the `presidio-analyzer` side, one would extend the [RemoteRecognizer](presidio-analyzer/analyzer/remote_recognizer.py) class and implement the network interface between `presidio-analyzer` and the endpoint of the model's container.
+
+### Adding new connectors
+
+Presidio's strength lies in its ability to be completely pluggable and customizable. Adding new types of databases, stream-analytics engines and data stores is extremely useful for many users.
+While we're working on adding more documentation on this type of customization, you can refer to these documents and resources:
+1. Scheduler
+
+   - The [scheduler documentation](docs/tutorial_scheduler.md) describes how to configure the data source and target for a periodic job
+
+2. Templates for source, target and stream configurations and service definitions, on the [presidio-genproto](https://github.com/microsoft/presidio-genproto) repo. For example: 
+
+	- [data sink protocols](https://github.com/microsoft/presidio-genproto/blob/master/src/datasink.proto). Specifically, the [DatasinkTypesEnum](https://github.com/microsoft/presidio-genproto/blob/1734e2635c253f79e4c44398315d92fe9d084601/src/datasink.proto#L37)
```

Comment:
any reason this is not master?

Reply:
It's a permalink, probably making this robust to changes in the target file. I can copy the original link with reference to the row but I guess that's the preferred way.